### PR TITLE
tslint deprecate warning

### PR DESCRIPTION
**Pull request by outofambit**
* adds a warning to the output of `yarn lint` when a method that has been [marked as deprecated](http://usejsdoc.org/tags-deprecated.html) is used
* uses `getDefaultRemote` as an example of deprecation

I'm also open to just deleting `getDefaultRemote` and its unit tests but I figured this deprecation mechanism might be useful for other situations in the future.

We can also make this an `error` in linting, if people would be open to that

**To Do:**
- [x] fix new linting errors

### CODE REVIEW

**Comment by outofambit** [comment]
@niik thanks for the feedback! i totally agree.

> Is there a reason why we're creating a new tslint.json here instead of reusing the one at the root of the repository?

this is so tslint only checks for deprecation errors in `app/src/`, but i think that putting it at the root makes more sense if we agree to not use deprecate rules in our own code

@shiftkey i'll take a look at this patch, thank you! i might put that in a separate PR, and let this one just deal with adding the deprecate rule to tslint

**Comment by outofambit** [comment]
see #5634

**Comment by outofambit** [comment]
thanks @shiftkey!

**Comment by shiftkey** [comment]
@outofambit is `help wanted` supposed to be on here?

**Comment by outofambit** [comment]
@shiftkey it was, but i went ahead and finished this :)

fixing the deprecation was easy since the error message clearly laid out how to fix it

**Comment by outofambit** [comment]
🍕

File: app/src/ui/resizable/resizable.tsx [hunk 1]
```diff
@@ -113,7 +113,7 @@ export class Resizable extends React.Component<IResizableProps, {}> {
   }
 }
 
-export interface IResizableProps extends React.Props<Resizable> {
+export interface IResizableProps extends React.ClassAttributes<Resizable> {
```
Comment (niik) [inline app/src/ui/resizable/resizable.tsx]: I think we should just drop the extends part here. This was from a time when we needed to get the `children` prop but these days that's automatically included by a type union.

**Comment by outofambit** [comment]
thanks @niik! 🍜

File: app/src/ui/resizable/resizable.tsx [hunk 2]
```diff
@@ -113,7 +113,7 @@ export class Resizable extends React.Component<IResizableProps, {}> {
   }
 }
 
-export interface IResizableProps extends React.Props<Resizable> {
+export interface IResizableProps {
```
Comment (iAmWillShepherd) [inline app/src/ui/resizable/resizable.tsx]: Is this change part of the work related to the deprecation warnings?

**Comment by outofambit** [comment]
@niik are we good to merge this?

**Pull request merged**